Implement Blog Data Model with Robust Migrations

DevKaranJ opened the pull request:
I've carefully crafted the database foundation for our blog app, ensuring a well-structured and scalable data model! ✨

**Here's a summary of the key features:**

- [x] **Rails Migrations Mastery:** Employed migrations flawlessly to create and manage the database schema, ensuring a clear and organized approach to database evolution. 🚀

- [x] **ERD Adherence:** Adhered strictly to the provided ERD diagram, guaranteeing accurate table and column definitions for seamless data organization. 📊

- [x] **Foreign Key Relationships:** Established robust foreign key relationships between tables, effectively enforcing data integrity and consistency. 🔗

- [x] **Indexing for Efficiency:** Created indexes for all foreign keys, optimizing query performance and data retrieval. ⚡

- [x] **User Photo Link:** Implemented the photo column in the users' table as a link to an image, enabling user personalization. 📸

- [x] **Best Practices Embraced:** Followed Ruby best practices and Gitflow workflow, ensuring code quality, maintainability, and collaboration efficiency. 🛠️

- [x] **Clear Documentation:** Provided comprehensive documentation, explaining the data model structure and design decisions for future reference and understanding. 📚

Ready for review 📝️

DevKaranJ committed:
Update database config

DevKaranJ committed:
Update database.yml

DevKaranJ committed:
Build Project Schema and migrate

DevKaranJ committed:
Installed linters

DevKaranJ committed:
Linters error fixed

DevKaranJ committed:
Update rubocop

DevKaranJ committed:
Updated readme.md

DevKaranJ committed:
Verified the migration status to ensure all migrations are up

TedLivist commented on db/migrate/20231227085329_create_comments.rb:
```diff
@@ -0,0 +1,12 @@
+  def change
+    create_table :comments do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :post, null: false, foreign_key: true
```
- [x] There is an error with your migration files setup. You are referencing `posts` here, but with the file arrangement, the posts table has not been created yet, leading to an error. The arrangement of your migration files should be such that the users table is created first, followed by the posts table, then the comments and likes, but you are creating the posts after creating comments, so the comments table would not be migrated correctly, because comment depends on the posts table.

  ![image](https://github.com/DevKaranJ/Blog/assets/36542986/e5b3222a-7e23-461d-8a84-0e2c5a53edc1)

TedLivist reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @DevKaranJ,

Good job so far!
✔️ Use of PostgreSQL for database
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

DevKaranJ committed:
Fix migration error: "relation 'comments' already exists"

Fixed issue where existing "comments" table caused migration error. Rolled back, restored missing files, and re-migrated successfully. Tables and schema now correct.

TedLivist reviewed (approved):
## STATUS: APPROVED ✔️ ✔️

Hi @DevKaranJ,

✔️ Correct implementation of ERD

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
